From e482aba14ff9d0b09c9fc074e07e97597a6b4e48 Mon Sep 17 00:00:00 2001
From: Wang Yufen <wangyufen@huawei.com>
Date: Mon, 21 Feb 2022 11:23:47 +0800
Subject: [PATCH 1009/2065] [Backport] skmsg: Lock ingress_skb when purging

mainline inclusion
from mainline-v5.13-rc1
commit 37f0e514db660f03f8982b8f4fbbd4b2740abe7d
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I545NW

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=37f0e514db660f03f8982b8f4fbbd4b2740abe7d

--------------------------------

Currently we purge the ingress_skb queue only when psock
refcnt goes down to 0, so locking the queue is not necessary,
but in order to be called during ->close, we have to lock it
here.

Signed-off-by: Cong Wang <cong.wang@bytedance.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Jakub Sitnicki <jakub@cloudflare.com>
Acked-by: John Fastabend <john.fastabend@gmail.com>
Link: https://lore.kernel.org/bpf/20210331023237.41094-2-xiyou.wangcong@gmail.com
Signed-off-by: Wang Yufen <wangyufen@huawei.com>
---
 net/core/skmsg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/skmsg.c b/net/core/skmsg.c
index 4ee4fe436847..f80428cc7264 100644
--- a/net/core/skmsg.c
+++ b/net/core/skmsg.c
@@ -638,7 +638,7 @@ void __sk_psock_purge_ingress_msg(struct sk_psock *psock)
 
 static void sk_psock_zap_ingress(struct sk_psock *psock)
 {
-	__skb_queue_purge(&psock->ingress_skb);
+	skb_queue_purge(&psock->ingress_skb);
 	__sk_psock_purge_ingress_msg(psock);
 }
 
-- 
2.17.1

